WIP: [RFC]: Bring Flask-GoogleMaps up to date with latest dependencies

argysamo opened the pull request:
Fixes #158

argysamo committed:
WIP: Update dependencies and functionality
- Update dependencies (Flask, mypy, python >3.8)
- Load javascript in an async manner
- Replace obsolete `Marker` with `AdvancedMarkerElement`

argysamo committed:
WIP: Functionality
- Introduce PinElement
- Minor javascript refactoring
- Remove icon functionality

argysamo committed:
WIP: Update javascript
- Replace javascript for loops with jinja. I did this in order to create "dynamic" js variables to have an "img" icon.
- This is not clever though. I will revert the js loop and then use jinja2 to create the img element when needed

argysamo committed:
WIP: Introduce Marker and MarkerContent
- MarkerContent is an abstract method. It can be either a Pin or Image
- Marker object comprises of MarkerContent
- Add unit tests
- I am bothered by the use of `eval` method for the content

argysamo commented:
hey @getcake  ,
I started working a feature branch to update Flask Google-Maps to use the latest Google Maps API and make it work with Flask 3.X.X.
Regarding the former, there's much more flexibility with creating markers. The colour of those can be easily customized via a [`PinElement`](https://developers.google.com/maps/documentation/javascript/advanced-markers/basic-customization). 
One could use an html `img` (or even a `div`) as a marker. I have (kind of) implemented the support for an `img` element. 

Making all these changes, it's kind of difficult to maintain backwards compatibility. 
This mainly regards, the way a user can define markers. 
More specifically, I wasn't able to think of way to maintain the `icon` notion with the new API, and thus I removed it. 

Along with the aforementioned updates, I am refactoring a bit the code introducing unit tests. 

Do you have any concerns? Do you think that these changes would be of value? 

Cheers,
Argyris

argysamo commented:
maybe @rochacbruno ?

argysamo committed:
WIP: Introduce marker_content_factory.py
- MarkerContentFactory returns the appropriate MarkerContent implementation (i.e. either Pin or Image)
- Add unit tests

argysamo committed:
WIP: Fix ci pipeline

- Reformat files with `black`
- Update dependencies. E.g. jsmin -> 3.0.1 as this version removes the use_2to3, which creates issues with latest setuptools

argysamo committed:
Bump dependencies

argysamo committed:
Move/refactor map tests

- Move and refactor test_map.py
- Bump Python dependency to 3.9
- Use latest flake8 revision

argysamo committed:
Update poetry version and ci.yaml

argysamo commented:
maybe @cuducos ?

argysamo commented:
@marcuxyz ?

rochacbruno commented:
LGTM @argysamo regarding the icon, as long as there is an alternative to add icons with the new markers API I am ok on doing a major release removing the icons API.

rochacbruno commented:
The example site build is failing because image is outdated on Netlify, it is running on @Riverfount account, I think what is needed is to check netlify panel and opt-in for an image update I guess.

Riverfount commented:
> The example site build is failing because image is outdated on Netlify, it is running on @Riverfount account, I think what is needed is to check netlify panel and opt-in for an image update I guess.

I can move it to another account, but which account?

Riverfount commented:
> > The example site build is failing because image is outdated on Netlify, it is running on @Riverfount account, I think what is needed is to check netlify panel and opt-in for an image update I guess.
> 
> I can move it to another account, but which account?

Sorry guys I can't move the site to another account on Netlify, but I can continue to maintain it in my account!

argysamo commented:
thanks so much for the feedback and the actions. Great to hear that you are interested in the new Markers API. I will continue with code change and testing, update the documentation and ping you here once the PR is ready to be reviewed!

marcuxyz commented:
it's very nice @argysamo . I believe that things as: 
 - packages updates
 - ci configurations 

Can be separated in another `pr`.

👏🏾 👏🏾 👏🏾 👏🏾 👏🏾

argysamo commented:
Hey @marcuxyz ,
Makes total sense. I will cherry pick right now the ci changes and create a new PR. 
Thanks!

rochacbruno commented:
needs a rebase

argysamo committed:
rebase

argysamo committed:
revert .pre-commit-config.yaml and remove icons.py

Riverfount commented:
Hi, guys if you need some help and just me I can do it, mark me on the comment because I see them more quickly.

argysamo committed:
Documentation and other updates

- Documentation: Describe the new markers setup
- Remove `label` field from marker.py
- Add support for `scale` for a pin.py

argysamo commented:
Hello all, 
I didn't make any progress for the last couple of weeks. 
I started working again on the documentation now, and will verify that the new changes don't break existing functionality. 
I will update you when changes are ready for review. 

Cheers,
Argyris

argysamo committed:
Update examples

- clusters, moving map, latlon don't work

getcake commented:
@argysamo Hey Argyrios & all — my apologies for the extensive inactivity on this repo— I greatly appreciate the continued support and drive to advance the extension further. 

I haven't ran through tests, but plan on coming out with a more streamlined workflow for testing commits like these to expedite PR's in the future. 

If you've written any code for this specific issue since then, I'd be happy to test & merge that as well.

(I know it's been a while though, so fully understand if that's a stretch) 

Thank you all again, and my apologies once more for the hiatus. Would love to get this thing off the ice.
